code/modules/reagents/reagent_containers/pill.dm

```diff
@@ -181,5 +181,17 @@
 	name = "speedy pill"
 	list_reagents = list("aranesp" = 10)
 
-
-
+/obj/item/reagent_containers/pill/floorpill
+	name = "floorpill"
+	desc = "A strange pill found in the depts of maintenance"
+	icon_state = "pill21"
+	var/static/list/names = list("maintenance pill","floorpill","mystery pill","suspicious pill","strange pill")
+	var/static/list/descs = list("Your feeling is telling you no, but...","Drugs are expensive, you can't afford not to eat any pills that you find."\
+	, "Surely, there's no way this could go bad.")
+
+/obj/item/reagent_containers/pill/floorpill/Initialize()
+	list_reagents = list(get_random_reagent_id() = rand(10,50))
+	name = pick(names)
+	if(prob(20))
+		desc = pick(descs)
+	. = ..()

```

Comment:
This should be the first line for consistency

Reply:
I need to change list_reagents before initializing, or it would be completely useless

Reply:
that's what this does...?

Reply:
. is the return value and ..() is what initialize would normally do
If I set . to ..() on the first line, it does initialize first
Running initialize before I can change reagent_list would cause the pill to be empty, that's why I change reagent_list before it can initialize

Reply:
`. = ..()` != `return ..()` except where you tell it to (or at the end of a return-less proc)

Reply:
Just do it and test it lol.

Reply:
oh also put `mapload` as the first argument to appease @optimumtact

Reply:
ok, follow me here
..() calls the parent proc. In the parent proc list_reagents is turned into the chemical contents of the pill
if I put ..() before I assign a value to list_reagents, the parent proc uses an empty list_reagents , wich is bad because then the pill is fucking empty
if I put ..() AFTER I change list_reagents, then the randomized list_reagents is used to decide the chemical contents
remember that . = ..() runs it RIGHT THEN, not when it actually returns, it just puts the value it returns into . and return that when we really return